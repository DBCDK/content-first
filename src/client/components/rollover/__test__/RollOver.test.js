import React from 'react';
import renderer from 'react-test-renderer';

import RollOver from '../RollOver.component';

jest.mock(
  '../../general/CheckmarkConnected.component',
  () => 'CheckmarkConnected'
);

const pid = '870970-basis:52038014';
const position = {x: 26, y: 46};
const description = 'lorem ipsum . . .';

const book = {
  pid: pid,
  creator: 'Nicole Boyle RÃ¸dtnes',
  title: 'test',
  cover: 'img/cover.png',
  description: description
};

describe('RollOver', () => {
  it('renders initial component', () => {
    const tree = renderer
      .create(
        <RollOver
          loading={true}
          position={position}
          description={description}
          onClick={() => {
            this.nextBook('next');
          }}
          book={book}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
