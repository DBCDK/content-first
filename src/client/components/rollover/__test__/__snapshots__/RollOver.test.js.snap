// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RollOver renders initial component 1`] = `
<div
  className="rollover-container "
  style={Object {}}
>
  <div
    className="rollover"
  >
    <img
      className="rollover-close"
      onClick={[Function]}
      src="/static/media/Kryds.e69a54ef.svg"
    />
    <div
      className="col-4 rollover-img"
    >
      <bookcover
        book={
          Object {
            "cover": "img/cover.png",
            "creator": "Nicole Boyle Rødtnes",
            "description": "lorem ipsum . . .",
            "pid": "870970-basis:52038014",
            "title": "test",
          }
        }
      />
    </div>
    <div
      className="col-8 text-left rollover-text"
    >
      <div
        className="col-12 rollover-title"
      >
        <h1>
          test
        </h1>
      </div>
      <div
        className="col-12 rollover-creator"
      >
        <h2>
          Nicole Boyle Rødtnes
        </h2>
      </div>
      <div
        className="col-12 rollover-description"
      >
        <p>
          lorem ipsum . . .. . .
        </p>
      </div>
      <div
        className="col-12"
      >
        <CheckmarkConnected
          book={
            Object {
              "book": Object {
                "pid": "870970-basis:52038014",
              },
            }
          }
          origin="Fra bogreol"
        />
      </div>
    </div>
    <div
      className="col-12 seperator-line"
    />
    <div
      className="col-12 rollover-bottom"
    >
      <span
        aria-hidden="true"
        class="glyphicon glyphicon-chevron-right"
        onClick={[Function]}
      />
      <span
        aria-hidden="true"
        class="glyphicon glyphicon-chevron-left"
        onClick={[Function]}
      />
    </div>
    <div
      className="clear"
    />
  </div>
</div>
`;
